### PR TITLE
[mono][jit] Optimize the experimental-gshared-mrgctx code

### PR DESCRIPTION
**Pull request by vargaz**
.

### CODE REVIEW

**Comment by vargaz** [comment]
This currently only affects platforms where the experimental-gshared-mrgctx option is true, which is wasm/ios.

**Comment by vargaz** [comment]
Failures are unrelated.

File: src/mono/mono/mini/mini-amd64.h [hunk 1]
```diff
@@ -459,6 +459,8 @@ typedef struct {
 #define MONO_ARCH_HAVE_INTERP_PINVOKE_TRAMP
 #define MONO_ARCH_HAVE_INTERP_ENTRY_TRAMPOLINE 1
 #define MONO_ARCH_HAVE_INTERP_NATIVE_TO_MANAGED 1
+// FIXME: Doesn't work on windows
+//#define MONO_ARCH_HAVE_INIT_MRGCTX 1
```
Comment (lambdageek) [inline src/mono/mono/mini/mini-amd64.h:463]: Why?

**Review by lambdageek — APPROVED** [review]

File: src/mono/mono/mini/mini-amd64.h (hunk 1 above)
Comment (vargaz) [reply src/mono/mono/mini/mini-amd64.h:463]: Causes test failures on CI if enabled.

Comment (lambdageek) [reply src/mono/mono/mini/mini-amd64.h:463]: That's kind of surprising.  Is the scratch register clobbering some real values for some reason? Or some calling convention problem?

Comment (vargaz) [reply src/mono/mono/mini/mini-amd64.h:463]: Couldn't reproduce locally, so just disabled it for now.

**Pull request merged**